### PR TITLE
Scaffolder/cookiecutter: Remove usage of mock-fs

### PR DESCRIPTION
**Pull request by anicke**
## Hey, I just made a Pull Request!

Removes usage of mock-fs from the tests in the "plugin-scaffolder-backend-module-cookiecutter" package. See https://github.com/backstage/backstage/issues/20436.

#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [X] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [X] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: plugins/scaffolder-backend-module-cookiecutter/src/actions/fetch/cookiecutter.test.ts [hunk 1]
```diff
@@ -106,36 +106,26 @@ describe('fetch:cookiecutter', () => {
       output: jest.fn(),
       createTemporaryDirectory: jest.fn().mockResolvedValue(mockTmpDir),
     };
-
-    // mock the temp directory
-    mockFs({ [mockTmpDir]: {} });
-    mockFs({ [`${join(mockTmpDir, 'template')}`]: {} });
+    mockDir.clear();
+    mockDir.addContent({ template: {} });
```
Comment (tudi2d) [inline plugins/scaffolder-backend-module-cookiecutter/src/actions/fetch/cookiecutter.test.ts:110]: Could this as well as all other usages just be done by `mockDir.setContent(...)`? What is the benefit of `addContent`? :)

File: plugins/scaffolder-backend-module-cookiecutter/src/actions/fetch/cookiecutter.test.ts [hunk 2]
```diff
@@ -47,6 +46,7 @@ jest.mock(
 );
 
 describe('fetch:cookiecutter', () => {
+  const mockDir = createMockDirectory();
```
Comment (tudi2d) [inline plugins/scaffolder-backend-module-cookiecutter/src/actions/fetch/cookiecutter.test.ts:49]: making sure that `mockDir.path` is equal to current `os.tmpdir()` implementation 

```suggestion
  const mockDir = createMockDirectory({ mockOsTmpDir: true });
```

**Review by tudi2d — CHANGES_REQUESTED** [review]
Tja @anicke - thanks for contributing! Looks good overall just some minor questions/clarifications needed. Heads-up that you will need to add a changeset. You can find a guide for that in the generated comment above & references to changesets in the related PRs removing `mock-fs` from other packages.

File: plugins/scaffolder-backend-module-cookiecutter/src/actions/fetch/cookiecutter.test.ts (hunk 1 above)
Comment (anicke) [reply plugins/scaffolder-backend-module-cookiecutter/src/actions/fetch/cookiecutter.test.ts]: Let's use setContent!

**Comment by anicke** [comment]
> Tja @anicke - thanks for contributing! Looks good overall just some minor questions/clarifications needed. Heads-up that you will need to add a changeset. You can find a guide for that in the generated comment above & references to changesets in the related PRs removing `mock-fs` from other packages.

Tjena! This had a changeset at one time but the guidelines says that changeset are not needed for "tests only" changes. https://github.com/backstage/backstage/pull/20668 does not have a changeset for example.

> Changesets are also not needed for changes that do not affect the published version of each package, for example changes to tests or in-line source code comments.

**Review by tudi2d — APPROVED** [review]
ah, sorry for correcting me - missed that :) 

awesome - looks good! 🚢

**Pull request merged**